Feat(eos_validate_state): custom fan & pwr states

chetryan opened the pull request:
## Change Summary

<!-- Enter short PR description -->

Allow custom states, more than just 'ok' for the
fan and power supply states in the hardware
validation task.

## Related Issue(s)

_No Issue found in github for this. This was raised internally._

## Component(s) name

`arista.avd.eos_validate`

## Proposed changes
<!--- Describe your changes in detail -->
<!--- Describe data model implemented for new features -->

## How to test
<!--- Please describe in detail how you tested your changes. -->
<!--- Include details of your testing environment, and the tests you ran to -->

_There is currently no Molecule tests for `eos_validate_state`._

## Checklist

### User Checklist

<!-- Add your own checklist using MD syntax and by replacing N/A -->
- [x] Tested on physical hardware. 
Models:
     - [x] 7808 with 1 power failure. The power failure was still reported when `notInserted` as was added to fan_states
     - [x] 7804 with some power fans missing. The failures were suppressed when `notInserted` was added to fan_states

### Repository Checklist

<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
- [x] My code has been rebased from devel before I start
- [x] I have read the [**CONTRIBUTING**](https://avd.sh/en/latest/docs/contribution/overview.html) document.
- [x] My change requires a change to the documentation and documentation have been updated accordingly.
- [x] I have updated [molecule CI](https://github.com/aristanetworks/ansible-avd/tree/devel/ansible_collections/arista/avd/molecule) testing accordingly. (check the box if not applicable)

chetryan committed:
Feat(eos_validate_state): custom fan & pwr states

Allow custome states, more than just 'ok' for the
fan and power supply states in the hardware
validation task.

carlbuchmann commented on ansible_collections/arista/avd/roles/eos_validate_state/defaults/main.yml:
```diff
@@ -19,6 +19,12 @@ fabric_name: "all"
 accepted_xcvr_manufacturers: "{{ validation_role.xcvr_own_manufacturers | arista.avd.default(['Arastra, Inc.', 'Arista Networks']) }}"
 
+# Allow different states for power supplies
+accepted_pwr_supply_states: "{{ validation_role.pwr_supply_states | arista.avd.default(['ok', 'Not Inserted']) }}"
```
Update the defaults to `OK` only - user must explicitly enable `Not Inserted` status.
```suggestion
accepted_pwr_supply_states: "{{ validation_role.pwr_supply_states | arista.avd.default(['ok']) }}"
```

Please also update README.md with a description of the use case.

chetryan replied:
Updated the main.yml.

carlbuchmann commented on ansible_collections/arista/avd/roles/eos_validate_state/defaults/main.yml:
```diff
@@ -19,6 +19,12 @@ fabric_name: "all"
+accepted_pwr_supply_states: "{{ validation_role.pwr_supply_states | arista.avd.default(['ok', 'Not Inserted']) }}"
+
+# Allow different states for fans
+accepted_fan_states: "{{ validation_role.fan_states | arista.avd.default(['ok', 'Not Inserted']) }}"
```
Update the defaults to OK only - the user must explicitly enable `Not Inserted``status.

```suggestion
accepted_fan_states: "{{ validation_role.fan_states | arista.avd.default(['ok']) }}"
```

Please also update README.md with a description of the use case.

chetryan replied:
Updated the main.yml.

carlbuchmann commented on ansible_collections/arista/avd/roles/eos_validate_state/README.md:
```diff
@@ -84,6 +84,13 @@ fabric_name: "all"
 accepted_xcvr_manufacturers: "{{ validation_role.xcvr_own_manufacturers | arista.avd.default(['Arastra, Inc.', 'Arista Networks']) }}"
 
+# Allow different states for power supplies
+accepted_pwr_supply_states: "{{ validation_role.pwr_supply_states | arista.avd.default(['ok', 'Not Inserted']) }}"
```
Update per comments in main.yml

chetryan replied:
Readme updated per main.yml and added use case description.

chetryan committed:
Fix(eos_validate_state): Revert ok as default

As discussed, the only default valid state
for power supplies and fans should be 'ok'.
If 'notInserted' needs to be allowed, it
should be done via the [fan|pwr_supply]_states.

chetryan committed:
Doc: Add description of use case

carlbuchmann reviewed (approved):
LGMT- Thanks for this fix @chetryan !
